### ISSUE TITLE
Feature: cosine distance with negative sampling

### ISSUE THREAD

**Issue by AnmolMago** [keras-team/keras#10424]
I am trying to train a model that uses cosine distance with negative samples. That is, the loss function is based on multiple values instead of a single y_true to y_pred comparison.

I noticed two things. One, there is no cosine distance with negative samples function for loss, as can be found in [CNTK (cosine_distance_with_negative_samples)](https://docs.microsoft.com/en-us/python/api/cntk.losses?view=cntk-py-2.5.1). Two, a loss function, including custom functions, must conform to the signature of LOSS(y_true, y_pred). I do not see any reason for this, as other libraries (such as Theano, CNTK and tensorflow) are able to use a loss function with multiple arguments.

I was wondering what was holding the implementation of these features back. If it is just because it is not a priority, I may go ahead and try to implement it myself.
